packages/taco-auth/src/providers/eip4361.ts

```diff
@@ -1,10 +1,13 @@
 import { ethers } from 'ethers';
 import { generateNonce, SiweMessage } from 'siwe';
+import { z } from 'zod';
 
+import { EIP4361_AUTH_METHOD } from '../auth-provider';
+import { AuthSignature } from '../auth-sig';
 import { LocalStorage } from '../storage';
-import { AuthSignature, EIP4361_AUTH_METHOD } from '../types';
 
-export type EIP4361TypedData = string;
+
+export const EIP4361TypedDataSchema = z.string();
```

Comment:
Should we use zod `refine` here as well? You can try calling the `SiweMessage` constructor as part of the refine function to ensure that the string can be parsed properly.

Reply:
Addressed here: [`f9e2030` (#527)](https://github.com/nucypher/taco-web/pull/527/commits/f9e203023327dc343e9aaf26fe2a362d5c8f2683)

Reply:
🎸